velox/expression/tests/ExpressionRunnerTest.cpp

```diff
@@ -181,9 +188,17 @@ int main(int argc, char** argv) {
     sql = restoreStringFromFile(FLAGS_sql_path.c_str());
     VELOX_CHECK(!sql.empty());
   }
+  if (FLAGS_engine == "spark") {
+    functions::sparksql::registerFunctions("");
+  } else if (FLAGS_engine == "presto") {
+    functions::prestosql::registerAllScalarFunctions();
+    aggregate::prestosql::registerAllAggregateFunctions();
+  } else {
+    std::cout << "Invalid value for --engine: " << FLAGS_engine << ". ";
```

Comment:
I think we have to use LOG(INFO) rather than cout here.

Reply:
Changing to LOG(INFO) means that by default when an incorrect value is passed executable exists without informing the user why unless they passed --v=1 and --logtostderr=1. 

I did modify this to model the other verifier for mode, an independant required argument:

```cpp
static bool validateMode(const char* flagName, const std::string& value) {
  static const std::unordered_set<std::string> kModes = {
      "common", "simplified", "verify", "query"};
  if (kModes.count(value) != 1) {
    std::cout << "Invalid value for --" << flagName << ": " << value << ". ";
    std::cout << "Valid values are: " << folly::join(", ", kModes) << "."
              << std::endl;
    return false;
  }

  return true;
}

static bool validateRegistry(const char* flagName, const std::string& value) {
  static const std::unordered_set<std::string> kRegistries = {
      "presto", "spark"};
  if (kRegistries.count(value) != 1) {
    std::cout << "Invalid value for --" << flagName << ": " << value << ". ";
    std::cout << "Valid values are: " << folly::join(", ", kRegistries) << "."
              << std::endl;
    return false;
  }
  if (value == "spark") {
    functions::sparksql::registerFunctions("");
  }
  else if (value == "presto") {
    functions::prestosql::registerAllScalarFunctions();
  }

  return true;
}

DEFINE_validator(mode, &validateMode);
DEFINE_validator(registry, &validateRegistry);
```

This also includes removing Presto aggregate registry.

Reply:
You can use LOG(ERROR/WARN) when we absolutely need to output error or warning messages etc. We typically use std::cout only in example programs.

Reply:
It seems reasonable to use std::cerr to print out the error message that explains why the program exited abnormally.

Reply:
The current standard in ExpressionRunner.cpp is to output flag misuse to std::cout. I'd be happy to modify the output for these as well if we prefer std::ceer.

https://github.com/facebookincubator/velox/blob/5fa64970fb9b6a9c7bf3103829c4132fc4f4942f/velox/expression/tests/ExpressionRunnerTest.cpp#L174-L177

https://github.com/facebookincubator/velox/blob/5fa64970fb9b6a9c7bf3103829c4132fc4f4942f/velox/expression/tests/ExpressionRunnerTest.cpp#L82-L93.